Lib/test/test_math.py

```diff
@@ -1006,6 +1007,21 @@ class T(tuple):
             self.assertEqual(math.dist(p, q), 5*scale)
             self.assertEqual(math.dist(q, p), 5*scale)
 
+    @support.cpython_only
+    @unittest.skipUnless(support.Py_DEBUG,
+                         '-X showrefcount requires a Python debug build')
+    def test_math_dist_leak(self):
+        # See https://github.com/python/cpython/issues/98897
+        code = textwrap.dedent("""
+        import math
+        try:
+            math.dist([1, 2], [3, 4, 5])
+        except ValueError:
+            pass
+        """)
+        res = script_helper.assert_python_ok('-X', 'showrefcount', '-c', code)
+        self.assertEqual(res.err, b'[0 refs, 0 blocks]\n')
```

Comment:
That's overkill. Python test suite has a built-in memory leak detected. No need to go throw a subprocess. Trying to run ``./python -m test test_math -m test_math_dist_leak -R 3:3`` without your fix.

I suggest:

```
with self.assertRaises(ValueError):
    math.dist([1, 2], [3, 4, 5])
```

Reply:
Right, I forgot about the built-in one :)